pkg/router/template/template_helper.go

```diff
@@ -175,7 +175,11 @@ func generateHAProxyCertConfigMap(td templateData) []string {
 		backendConfig := backendConfig(string(k), cfg, hascert)
 		if entry := haproxyutil.GenerateMapEntry(certConfigMap, backendConfig); entry != nil {
 			fqCertPath := path.Join(td.WorkingDir, certDir, entry.Key)
-			lines = append(lines, strings.Join([]string{fqCertPath, entry.SSLBindConfig, entry.Value}, " "))
+			if td.DisableHTTP2 {
```

Comment:
@frobware does it make sense to update the `TestGenerateHAProxyMap()` unit test to support this new logic? cc: @Miciah

Reply:
If there's a unit test, then definitely.

Reply:
Done in https://github.com/openshift/router/pull/133/commits/b0767d17c8b29269a303d75cd0e9c6336a9eb482